feat: add DATADOG_STORAGE_TIER env support for logs tools

j8d opened the pull request:
## Summary
Adds support for the `DATADOG_STORAGE_TIER` environment variable to enable querying logs from Datadog's Flex tier storage.

## Problem
On November 17, 2025, we moved to the Flex tier. The current MCP server has no way to specify which storage tier to query, so all log searches default to the indexes tier and return no results for Flex-only logs.

## Solution
- Read `DATADOG_STORAGE_TIER` environment variable at module initialization
- Validate against supported values: `indexes`, `online-archives`, `flex`
- Pass `storageTier` to both `get_logs` and `get_all_services` Datadog API calls when configured
- Fully backwards compatible - if env var is not set, behavior is unchanged

## Usage
Set in MCP config:
```json
{
  "mcpServers": {
    "mcp-server-datadog": {
      "env": {
        "DATADOG_STORAGE_TIER": "flex"
      }
    }
  }
}

Commit:
feat: add DATADOG_STORAGE_TIER env support for logs tools

winor30 commented:
@j8d 
Thanks for this PR! The implementation looks good.

One request before merging:
Please add tests for the `DATADOG_STORAGE_TIER` functionality in `tests/tools/logs.test.ts`.

j8d committed:
Merge branch 'main' into feat/datadog-storage-tier-env

Commit:
fix: correct storageTier serialization in logs API requests

- Refactor filter object construction in get_logs and get_all_services
- Update tests to check for 'storage_tier' (snake_case) in request body
  as the Datadog API client serializes storageTier to storage_tier

Commit:
docs: remove environment-specific comment from README

Commit:
refactor: move DATADOG_STORAGE_TIER tests to separate file

Move the DATADOG_STORAGE_TIER environment variable tests from
logs.test.ts to a dedicated logs-storage-tier.test.ts file to
prevent MSW server conflicts during concurrent test execution.

winor30 reviewed (approved):
thank you for your contribution!
